File: api/src/main/java/com/ruby/rubia_server/auth/AuthResponse.java
package com.ruby.rubia_server.auth;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class AuthResponse {
    private String token;
    private UserInfo user;
    private long expiresIn;
    private String companyId;
    private String companySlug;
}

File: api-dotnet/Rubia.Server/DTOs/AuthResponse.cs
namespace Rubia.Server.DTOs;

public class AuthResponse
{
    public string Token { get; set; } = string.Empty;
    public long ExpiresIn { get; set; }
    public string TokenType { get; set; } = "Bearer";
    public bool RequiresWhatsAppSetup { get; set; }
    public UserInfo User { get; set; } = new();
}

public class UserInfo
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Role { get; set; } = string.Empty;
    public Guid CompanyId { get; set; }
    public string CompanySlug { get; set; } = string.Empty;
    public Guid CompanyGroupId { get; set; }
    public Guid DepartmentId { get; set; }
}